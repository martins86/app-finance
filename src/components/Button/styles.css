.btn {
    border-radius: 0.2rem;
    font-weight: 600;
    font-size: 1rem;
    padding: 1rem 1.6rem;
    cursor: pointer;
}

.btn--full-width {
    max-width: 350px;
    width: 100%;
}

.btn--primary {
    border: none;
    color: #ffffff;
    background-color: #046ee5;
}

.btn--primary:hover {
    color: #f4f4f4;
    background-color: #0263cf;
}

.btn--primary:disabled {
    color: #d4d4d4;
    background-color: #85a2c3;
    cursor: not-allowed;
}