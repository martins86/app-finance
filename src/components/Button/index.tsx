import './styles.css';

export interface ButtonParams {
  className?: string;
  onClick?: any;
  disabled?: boolean;
  text?: string;
}

const Button = ({ className, onClick, disabled, text }: ButtonParams) => {
  return (
    <button
      type="button"
      data-testid="button"
      className={className}
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </button>
  );
};

export default Button;
