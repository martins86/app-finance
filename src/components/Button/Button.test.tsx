import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Button from '.';

describe('Test <Button />', () => {
    test('Should render the button with the text "Any Text"', () => {
        // Arrange
        render(<Button text="Any Text" />);

        // Act
        const elButton = screen.getByRole('button', { name: /Any Text/i });

        // Assert
        expect(elButton).toBeInTheDocument();
        expect.assertions(1);
    });

    test('Should call function on button click', () => {
        // Arrange
        const fn = jest.fn();
        render(<Button text="Any Text" onClick={fn} />);

        // Act
        const elButton = screen.getByRole('button', { name: /Any Text/i });
        userEvent.click(elButton);

        // Assert
        expect(fn).toBeCalledTimes(1);
    });

    test('Should be disabled when disabled is true', () => {
        // Arrange
        render(<Button text="Any Text" disabled={true} />);

        // Act
        const elButton = screen.getByRole('button', { name: /Any Text/i });

        // Assert
        expect(elButton).toBeDisabled();
    });

    test('Should be enabled when disabled is false', () => {
        // Arrange
        render(<Button text="Any Text" disabled={false} />);

        // Act
        const elButton = screen.getByRole('button', { name: /Any Text/i });

        // Assert
        expect(elButton).toBeEnabled();
    });

    test('Should be enabled when disabled send', () => {
        // Arrange
        render(<Button text="Any Text" />);

        // Act
        const elButton = screen.getByRole('button', { name: /Any Text/i });

        // Assert
        expect(elButton).toBeEnabled();
    });
});
