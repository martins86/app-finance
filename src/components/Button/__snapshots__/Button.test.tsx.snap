// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button /> Should match snapshot 1`] = `
<div>
  <button
    class="any-class-style"
    data-testid="button"
    title="Any Text"
    type="button"
  >
    Any Text
  </button>
</div>
`;
